torch/custom_class.h

```diff
@@ -58,16 +61,21 @@ class class_ {
 
     // We currently represent custom classes as torchscript classes with a
     // capsule attribute
-    classTypePtr =
-        ClassType::create(c10::QualifiedName(qualClassName), classCU());
-    classTypePtr->addAttribute("capsule", CapsuleType::get());
-
-    c10::getCustomClassTypeMap().insert({typeid(c10::intrusive_ptr<CurClass>).name(),
-                              c10::StrongTypePtr(classCU(), classTypePtr)});
-    c10::getCustomClassTypeMap().insert({typeid(c10::tagged_capsule<CurClass>).name(),
-                              c10::StrongTypePtr(classCU(), classTypePtr)});
-
-    classCU()->register_type(classTypePtr);
+    classTypePtr = at::ClassType::create(
+        c10::QualifiedName(qualClassName),
+        std::weak_ptr<script::CompilationUnit>());
+    classTypePtr->addAttribute("capsule", at::CapsuleType::get());
+
+    c10::getCustomClassTypeMap().insert(
+        {typeid(c10::intrusive_ptr<CurClass>).name(),
+         c10::StrongTypePtr(
+             std::shared_ptr<script::CompilationUnit>(), classTypePtr)});
+    c10::getCustomClassTypeMap().insert(
+        {typeid(c10::tagged_capsule<CurClass>).name(),
+         c10::StrongTypePtr(
```

Comment:
If the class map also has null CompilationUnits, can we just store name -> ClassType directly?